### PR TITLE
Add task-list component.

### PR DESCRIPTION
**Pull request by marineb**
# What's in this PR?

This is not done, just creating a PR to make my progress transparent. 

## References
Fixes #851

Progress on: #841

### CODE REVIEW

**Comment by JoshSmith** [comment]
@marineb looking good! I have some changes from #859 and also some things locally that can help this along, too. I'll comment with what I have tomorrow.

Didn't fully finish tonight (couldn't figure out a good way to hook into the Ember lifecycle with this library), but here's what I have:

**Comment by marineb** [comment]
Sounds good. Yes I saw that. It looks great!

**Comment by WenInCode** [comment]
@marineb @JoshSmith I added a simple page-object, and a few tests, and fixed a few eslint issues.

One thing I wanted to do, but was having some difficulty with, was test the ordering of tasks. On the page-object I have a collection of `TaskCard` page objects. But I am unable to access the taskCard properties from the task-list tests. I was trying it via `page.taskCards(1).number.text` but number is undefined. 

Time to sleep though, just thought I should update you both.

File: tests/pages/components/task-list.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+  collection,
+  text
+} from 'ember-cli-page-object';
+
+import TaskCard from 'code-corps-ember/tests/pages/components/task-card';
+
+export default {
+  title: text('[data-test-selector="task-list name"]'),
+
+  taskCards: collection(TaskCard)
```
Comment (JoshSmith) [inline tests/pages/components/task-list.js]: I've not seen this syntax before, so I'm suspecting this is the issue you're having.

**Comment by marineb** [comment]
It looks great!

File: tests/pages/components/task-list.js (hunk 1 above)
Comment (WenInCode) [inline tests/pages/components/task-list.js]: I figured you can create a collection with
```
collection({
  scope: 'selector',
  attributes...
});
```
so maybe putting the imported `TaskCard` page definition in there might work.

**Comment by JoshSmith** [comment]
@WenInCode were you planning to add a test for order still?

**Comment by JoshSmith** [comment]
🙌  @marineb @WenInCode

**Pull request merged**